File: examples/advanced-crud/src/main/java/org/example/advancedcrud/app/entity/MovieActorId.java
package org.example.advancedcrud.app.entity;

import org.hibernate.Hibernate;

import java.io.Serializable;
import java.util.Objects;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Embeddable
public class MovieActorId implements Serializable {
    private static final long serialVersionUID = 6615578765469384867L;
    @Column(name = "movie_id", nullable = false)
    private Integer movieId;

    @Column(name = "actor_id", nullable = false)
    private Integer actorId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        MovieActorId entity = (MovieActorId) o;
        return Objects.equals(this.actorId, entity.actorId) &&
                Objects.equals(this.movieId, entity.movieId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(actorId, movieId);
    }

}

File: examples/basic-crud-kotlin/src/main/kotlin/org/example/basiccrudkotlin/app/entity/MovieActorId.kt
package org.example.basiccrudkotlin.app.entity

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import org.hibernate.Hibernate
import java.io.Serializable
import java.util.*

@Embeddable
open class MovieActorId : Serializable {
    @Column(name = "MOVIE_ID", nullable = false)
    open var movieId: Int? = null

    @Column(name = "ACTOR_ID", nullable = false)
    open var actorId: Int? = null
    override fun hashCode(): Int = Objects.hash(movieId, actorId)
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false

        other as MovieActorId

        return movieId == other.movieId &&
                actorId == other.actorId
    }

    companion object {
        private const val serialVersionUID = -7721510229263762687L
    }
}